_posts/2015-02-16-Utilising-API-Blueprint-in-API-Clients.md

```diff
@@ -0,0 +1,120 @@
+
+```markdown
+# Polls
+Polls is a simple web service that allows consumers to view polls and vote in them.
+
+## Group Question
+Resource related to questions in the API.
+
+## Question [/questions/{question_id}]
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+
+### View a question detail [GET]
+
+## Choice [/questions/{question_id}/choices/{choice_id}]
+
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+    + choice_id (number) ... ID of the Choice in form of an integer
+
+### Vote [PUT]
+
+This action allows you to vote on a question's choice.
+
+## Questions collection [/questions{?page}]
++ Parameters
+    + page (optional, number) ... The page of questions to return
+
+### List all questions [GET]
+### Create a new question [POST]
+```
+
+We can pull out any of the actions from our blueprint by the resource and action name in Swift:
+
+```swift
+if let transition = blueprint?.transition("Choice", action: "Vote") {
+}
+```
+
+This transition allows us to introspect values at run-time, retrieving protocol information such as the HTTP method and URI along with all the URI parameters.
+
+```swift
+println(transition.uri)
+println(transition.method)
+
+for (name, parameter) in transition.parameters {
+  println(name)
+  println("\(parameter.value) \(parameter.defaultValue)")
+}
+```
+
+We can use the elegant Swift HTTP networking library [Alamofire](https://github.com/Alamofire/Alamofire) along with some convenience methods we’ve built to perform HTTP requests from our transition directly.
+
+```swift
+let baseURL = NSURL(string: "http://polls.apiblueprint.org/")
+let poll = [
+  "title": "What's your favourite programming language?"
+]
+
+request(baseURL, transition, attributes:poll).responseJSON { (_, _, JSON, _) in
+  println(JSON)
+}
+```
+
+For the other action, to vote on a question by it's ID. We have a URI parameter
+`id` and `choice_id`. We can substitute this into the URI and make the request like so:
+
+```swift
+let params = [
+  "id": 1,
+  "choice_id": 2,
+]
+
+request(baseURL, transition, parameters:params).responseJSON { (_, _, JSON, _) in
+  println(JSON)
+}
+```
+
+This will result in performing a PUT request to `/questions/1/choices/2`.
+
+If you’re interested in giving this a go, check out our [installation guide](https://github.com/the-hypermedia-project/representor-swift/wiki/API-Blueprint) on using API Blueprint in Swift.
+
+Of course, we’ve integrated this into our existing code examples. So right from Apiary you can see how it looks to perform a transition in Swift.
+
+<img src="/images/2015-02-16-Utilising-API-Blueprint-in-API-Clients/code-example.png" width="665" height="364" alt="API Blueprint Representor Code Examples" />
+
+We would love to hear your thoughts in the comments section below.
```

Comment:
The only thing I think missing below is too put and else block with a comment about what you would do to recover or alternate path for loose coupling. E.g.

```swift
if transition //...
  // your current implementation
else
  // Gracefully handle the absence of the transition (or something like this)
end
```